tcms/kiwi_auth/tests/test_views.py

```diff
@@ -199,6 +201,25 @@ def test_register_user_and_activate_by_admin(self):
                                 '<a href="mailto:{}">{}</a>'.format(email, name),
                                 html=True)
 
+    def test_invalid_form(self):
+        response = self.client.post(
+            self.register_url,
+            {
+                "username": "kiwi-tester",
+                "password1": "password-1",
+                "password2": "password-2",
+                "email": "new-tester@example.com",
+            },
+            follow=False,
+        )
+
+        self.assertEqual(
+            response.context['form'].error_messages,
+            {'password_mismatch': 'The two password fields didn’t match.'}
+        )
```

Comment:
It is better to use `assertContains()` which makes sure that the page is actually showing the message.